### ISSUE TITLE
Turn off function name warnings for methods implemented in external interfaces

### ISSUE THREAD

**Issue by ncw** [golang/lint#172]
I'm implementing as interface defined in an external package.

Unfortunately the names in the external package interface don't conform to `golint` standards.

Could golint work out that I'm implementing a method in an interface in an external package and not warn about it?

### External library with well established (non golint friendly) method name. ###

```go
package external

type Authenticator interface {
	CdnUrl() string
}
```

### My code which I can change, but I can't ever get rid of the golint message  ###

```go
package mypackage

import "external"

type auth struct {
}

// golint gives the warning "func CdnUrl should be CdnURL" here
// which this package can't do anything about
func (a *auth) CdnUrl() string {
	return ""
}

// Check the interfaces are satisfied
var _ external.Authenticator = (*auth)(nil)
```

**Comment by dsymonds** [comment]
In general it can't, sorry.

**Comment by ncw** [comment]
That's what I thought you'd say!

Thanks for looking into it. 

Nick
